pii-manager/src/pii_manager/lang/any/credit_card.py

```diff
@@ -0,0 +1,55 @@
+"""
+Find valid credit card numbers:
+1. Obtain candidates, by using a generic regex expression
+2. Validate candidates by
+    - using a more exact regex
+    - validating the number through the Luhn algorithm
+"""
+
+import re
+
+from stdnum import luhn
+
+from pii_manager import PiiEnum, PiiEntity
+from pii_manager.helper import BasePiiTask
+
+
+# ----------------------------------------------------------------------------
+
+# regex for credit card type
+# https://www.regular-expressions.info/creditcard.html
+_CREDIT_PATTERN = r"""4[0-9]{12}(?:[0-9]{3})? |
+                      (?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12} |
+                      3[47][0-9]{13} |
+                      3(?:0[0-5]|[68][0-9])[0-9]{11} |
+                      6(?:011|5[0-9]{2})[0-9]{12} |
+                      (?:2131|1800|35\d{3})\d{11}"""
+
+
+_REGEX_CREDIT_CARD = None
+
+
+class CreditCard(BasePiiTask):
+    """Credit card numbers for most international credit cards (recognize & validate)"""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        # Compile the big credit card regex
+        global _REGEX_CREDIT_CARD
+        if _REGEX_CREDIT_CARD is None:
+            _REGEX_CREDIT_CARD = re.compile(_CREDIT_PATTERN, flags=re.VERBOSE)
+
+    def find(self, doc: str):
+        # First find candidates
+        for cc in re.finditer(r"\b \d (?:\d[ -]?){14} \d \b", doc, flags=re.X):
```

Comment:
This regex is not compiled?

Reply:
> Nice work, well designed and like the effort to unit tests. Still some efforts needed to gather more tasks for all languages and ensure all the process is ok with non-western lanagues, but a good progressin structure. One question though: this is a duplicate of your repo, why not using it as an external library?

The idea is indeed for it to be a library, so that code using it just adds it as a dependency (which allows us to version in) and pip-install it. But I got the impression that @ontocord  wanted to merge all tools into this repository, hence the PR. It will still be a separate folder that generates an independent package.